bin/automerge-prs

```diff
@@ -79,12 +91,20 @@ EOM
   exit 0
 fi
 
+read -r defaultBranch < \
+  <(gh repo view "$GH_REPO" --json defaultBranchRef --jq '.defaultBranchRef.name')
+
+gh api \
+  "/repos/$GH_REPO/branches/$defaultBranch/protection" \
+  --jq '.required_status_checks.contexts[]' |
+  sort >"$tmp/required-statuses.txt"
```

Comment:
Hmm, so this requires a permission that I'm not sure Actions is able to even request.

That might be a deal-breaker for this whole idea.

Reply:
I think the choices are:

1. As for the list of required statuses as an input, or
2. Do this if _any_ status checks fail

Reply:
Is there some way to _attempt_ a merge, which would be rejected if a required status is failed?

Reply:
That's a good idea. I don't know what happens if a bot tries to merge around branch protections, I guess we'd have to test it.